src/models/tools/tool-content-info.ts

```diff
@@ -51,29 +50,24 @@ interface IToolContentInfoMap {
   [id: string]: IToolContentInfo;
 }
 const gToolContentInfoMapById: IToolContentInfoMap = {};
-const gToolContentInfoMapByTool: IToolContentInfoMap = {};
 
 export function registerToolContentInfo(toolContentInfo: IToolContentInfo) {
-  gToolContentInfoMapById[toolContentInfo.id] = toolContentInfo;
-  gToolContentInfoMapByTool[toolContentInfo.tool] = toolContentInfo;
+  // toLowerCase() for legacy support of tool names
+  gToolContentInfoMapById[toolContentInfo.id.toLowerCase()] = toolContentInfo;
 }
 
 // ToolContent id, e.g. kDrawingToolID, kGeometryToolID, etc.
 export function getToolContentInfoById(id: string) {
-  return gToolContentInfoMapById[id];
-}
-
-// tool name used in a few places, e.g. "drawing", "geometry", etc.
-export function getToolContentInfoByTool(tool: string) {
-  return gToolContentInfoMapByTool[tool];
+  // toLowerCase() for legacy support of tool names
+  return id ? gToolContentInfoMapById[id.toLowerCase()] : undefined;
 }
 
 export function getToolContentModels() {
   return Object.values(gToolContentInfoMapById).map(info => info.modelClass);
 }
 
 export function getToolIds() {
-  return Object.keys(gToolContentInfoMapById);
+  return Object.keys(gToolContentInfoMapById).map(toolId => gToolContentInfoMapById[toolId].id);
```

Comment:
Is the extra look up necessary since the list of keys is already the list of ids?

Reply:
No, I reflexively use `Object.keys()` instead of `Object.values()` because the latter requires a more recent compilation target that isn't always available (e.g. DESE), but in this case clearly we're already using `Object.values()` so being consistent would be preferable. I'll make the change.

Reply:
And the other point, which may actually be what you were asking about, is that the keys are `toLowerCase`d, so we need to look up the actual id.

Reply:
Looks good.